src/coprocessor/dag/executor/aggregation.rs

```diff
@@ -69,26 +70,26 @@ impl AggrFunc {
     }
 }
 
-pub struct AggregationExecutor<'a> {
+pub struct AggregationExecutor {
     group_by: Vec<Expression>,
     aggr_func: Vec<AggrFuncExpr>,
-    group_keys: Vec<Rc<Vec<u8>>>,
-    group_key_aggrs: HashMap<Rc<Vec<u8>>, Vec<Box<AggrFunc>>>,
+    group_keys: Vec<Arc<Vec<u8>>>,
+    group_key_aggrs: HashMap<Arc<Vec<u8>>, Vec<Box<AggrFunc>>>,
```

Comment:
I think you can use `OrderMap` here.

Reply:
Do you mean builtin `BTreeMap`, or other `OrderMap` in third-party crates?

Reply:
I find `bluss/ordermap`, which has a method `pop`. I think it's good.